samples/XCT.Sample/Helpers/XLog.cs

```diff
@@ -121,7 +121,7 @@ static string FormatLogString(object data = null, string tag = null, string memb
 				line.Append(dataString);
 			}
 
-			if (!string.IsNullOrEmpty(sourceFilePath))
+			if (sourceFilePath != null && !string.IsNullOrEmpty(sourceFilePath))
```

Comment:
Yup - you're right: this is done to appease Roslyn.

I'd rather use this pattern than implement `!.` anywhere because that operator can lead to bugs in the future. For example, if the code around this changes, and `!.` is not removed, then we'll be hiding possible null references